tests/test_Step.cc

```diff
@@ -497,7 +497,7 @@ TEST_CASE("execute(): Immediate termination", "[Step]")
 
     REQUIRE_THROWS_AS(step.execute(context, &comm, 0), Error);
     REQUIRE(comm.immediate_termination_requested_);
-    REQUIRE(std::get<long long>(context.variables["a"]) == 0LL);
+    REQUIRE(std::get<long long>(context.variables["a"]) == -1LL);
```

Comment:
Unfortunately I did not touch this [test case](https://github.com/taskolib/taskolib/issues/24#issuecomment-1293204205), see [note](https://github.com/taskolib/taskolib/issues/24#issuecomment-1293204205) 🙂 . Anyhow I would not use any `Context` variable. My proposal is to simplify it without handling any variables (... because we wanted to test `comm.immediate_termination_requested_` and that's it) :

```c++
TEST_CASE("execute(): Immediate termination", "[Step]")
{
    Context context;

    Step step;
    CommChannel comm;

    context.lua_init_function =
        [&comm](sol::state& sol)
        {
            sol["request_termination"] =
                [&comm]
                {
                    comm.immediate_termination_requested_ = true;
                };
        };

    // The script needs a certain number of steps to make sure the termination
    // condition is even tested
    step.set_script("request_termination(); for a = 1, 1000 do end; return true");

    REQUIRE_THROWS_WITH(step.execute(context, &comm, 0), Contains("Stop on user request"));
    REQUIRE(comm.immediate_termination_requested_);
}
```

Reply:
Well, I find the changed test useful, because it tests if variables are synced back on `immediate_termination_requested_`, which is part of `immediate_termination_requested_` I guess ;)

Reply:
Regarding the test... you are right, sorry. Why did I think that? /shrug

![image](https://user-images.githubusercontent.com/16012374/198571176-05d114fd-1fa2-472d-8909-766fdc4e59db.png)

Reply:
I would propose to focus on `CommChannel::immediate_termination_requested_` and not on variables in this test case. I thought that this is the sense of test cases - small pieces of code that tests one or two aspects. Checking variables and its states is another test case. So: simplify the test case.

Sadly I must say that I am also shooting myself in the food because some of my test cases are to complex somehow... _simplification_ is the word of the day 😉 .

Reply:
Well, but we do test that variables are accessible - in normal cases - elsewhere.

Here is a specific check that that still holds id execution is interrupted via immediate-termination-req.

For me the test and the combination makes perfect sense. What I could not grasp is why the test tested the variable to ZERO. That is somehow unexpected. Either I want to test the variables and *I* would expect the variable to be -1; but that would have resulted in more code changes and finding the reason, OR this is a case of 'hmm, somehow unexpected but that is how it is' handwaving, shrugging, and just comparing to zero.

Either I do not test it, or I DO test it and see that the value is not as expected. Why test and require a unexpected result :grimacing: 
That was the reason for the question: Is there some hindsight why the unexpected zero has been expected and checked here.

Well, I guess only @alt-graph  can answer that.

Whatever. Lets focus on the actual change in the PR :-D